### ISSUE TITLE
0.4.19 RPM not installable on OpenSuse

### ISSUE THREAD

**Issue by tuxtard** [nylas/N1#1832]
Problem: nothing provides gir1.2-gnomekeyring-1.0 needed by nylas-0.4.19-0.1.x86_64

Previous versions are installable.

**Comment by Chowlz** [comment]
Similar situation with Fedora 23 

Problem: nothing provides libgnome-keyring0

**Comment by dmuhs** [comment]
I figure with Fedora 23 this is a simple package naming issue. By default the following packages are provided and installable with dnf:
```
libgnome-keyring-3.12.0-5.fc23.i686        
libgnome-keyring-3.12.0-5.fc23.x86_64      
libgnome-keyring-devel-3.12.0-5.fc23.i686  
libgnome-keyring-devel-3.12.0-5.fc23.x86_64
```

**Comment by Bergg** [comment]
Exact same issue on F23 , with libgnome-keyring and libgnome-keyring-devel installed

sudo dnf install N1.rpm 
Last metadata expiration check: 2:15:56 ago on Wed Mar 30 14:10:40 2016.
Error: nothing provides libgnome-keyring0 needed by nylas-0.4.19-0.1.x86_64
(try to add '--allowerasing' to command line to replace conflicting packages)

**Comment by hnq90** [comment]
I got this issue on Fedora 22 :+1:

**Comment by Crosant** [comment]
Can confirm this Issue on Fedora 23

**Comment by Evolution169** [comment]
Same here receiving this from terminal on vanilla Fedora 23
```
error: Failed dependencies:
	libgnome-keyring0 is needed by nylas-0.4.19-0.1.x86_64
	gir1.2-gnomekeyring-1.0 is needed by nylas-0.4.19-0.1.x86_64
```

**Comment by PI-Victor** [comment]
bump for interest. i can't update the damn thing

fedora 23 here.

**Comment by dtoebe** [comment]
bump for openSUSE (Tumbleweed)

**Comment by bengotow** [comment]
Hey folks—it sounds like we need to update the configuration here: https://github.com/nylas/N1/blob/master/build/resources/linux/redhat/nylas.spec.in#L9 to the versions which are available on Fedora 23 / openSUSE and see if it works. We added these dependencies so that the app could store it's API keys in the system keychain using this library:

https://github.com/atom/node-keytar

They note: `If you are using a Red Hat-based system you need to run sudo yum install libgnome-keyring-devel.`. Would that be a workable solution?

**Comment by mbilker** [comment]
Hmm. I wonder if instead of that we use the require on the `.so` file like a normal package would do. I will look into this further on an actual Fedora VM.

**Comment by PI-Victor** [comment]
@bengotow nylas's rpm dependencies right now are for debian packages, regardless of what we install on our systems to make that lib work, the nylas rpm still expects packages that don't exist named that way on our systems. it will only work after you update your package dependencies to reflect the ones for fedora.

i'd submit a PR, but i can't find the equivalent of gir1.2* for fedora ? i completely hate package naming differences on linux. :/

**Comment by Cris70** [comment]
Same here (OpenSUSE Tumbleweed).
However, by ignoring the dependency, I can install N1 and it works correctly.

**Comment by Bergg** [comment]
Not sure how to ignore dependencies with dnf, so here is what worked for me
sudo rpm -Uvh --nodeps N1.rpm

**Comment by dyskette** [comment]
Probably something like this in the .spec file:

```
Requires:       libgnome-keyring.so.0
%if !0%{?fedora}
Requires:       typelib(GnomeKeyring)
%endif
```

**Comment by mbilker** [comment]
@dyskette That would be good, except under the current build system the RPMs are made under Ubuntu. It would be preferred not to have to provide separate RPMs for each distribution.

**Comment by 1dot75cm** [comment]
Hi, guys.
I use [Copr](https://copr.fedorainfracloud.org/coprs/mosquito/n1/) to build RPM for Fedora/RHEL7. It's work fine. @dyskette  @Bergg @PI-Victor 

It has not been tested for openSUSE. @Cris70 @dtoebe

**Comment by albel1205** [comment]
> Not sure how to ignore dependencies with dnf, so here is what worked for me
sudo rpm -Uvh --nodeps N1.rpm

This way just worked for me. I also wonder that what was these dependencies using for?

**Comment by PI-Victor** [comment]
i went back to evolution. i see there's a lack of care for this app and will use it again once it picks up.

**Comment by RayFoss** [comment]
@Ionut, It's really easy to build N1 and put a desktop file with "electron
app" as exec. Which is what I did.

The RPM is also just a wrapper of precompiled binaries, so running it
straight off after extracting it should just work too. N1 is far better
than evolution imo.

On Fri, Apr 22, 2016 at 1:49 PM Ionut Palade <notifications@github.com>
wrote:

> i went back to evolution. i see there's a lack of care for this app and
> will use it again once it picks up.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/nylas/N1/issues/1832#issuecomment-213527811>
>
-- 

*Ray Foss*
M: (707) 739-6253 P: (315) 487-2111 F: (315) 487-4208

**Comment by biji** [comment]
Would you please make xdg-app bundle for N1 ;)

**Comment by jpeeler** [comment]
From what I can tell, it appears that the requires should be the libgnome-keyring library and development packages.

$ dnf whatprovides libgnome-keyring.so.0
Last metadata expiration check: 0:13:59 ago on Thu Apr 28 17:06:31 2016.
libgnome-keyring-3.12.0-5.fc23.i686 : Framework for managing passwords and other secrets
Repo        : fedora

$ dnf whatprovides /usr/lib64/girepository-1.0/GnomeKeyring-1.0.typelib
Last metadata expiration check: 0:15:22 ago on Thu Apr 28 17:06:31 2016.
libgnome-keyring-3.12.0-5.fc23.x86_64 : Framework for managing passwords and other secrets
Repo        : @System

http://packages.ubuntu.com/precise/amd64/gir1.2-gnomekeyring-1.0/filelist
http://packages.ubuntu.com/precise/amd64/libgnome-keyring0/filelist

node-keytar lists needing libgnome-keyring-devel, which makes sense since it sounds like compiling is necessary. Just for complete information:

$ dnf repoquery -q --list libgnome-keyring-devel
http://paste.fedoraproject.org/360848/18789291/

Hope that's helpful.

**Comment by RayFoss** [comment]
To install on fedora 22+ from mosquito's repository `sudo dnf copr enable mosquito/n1`

**Comment by mbilker** [comment]
This is fixed in the latest version with the `libgnome-keyring` dependency fixed.

**Comment by mbilker** [comment]
Should work. Please post the full command output here if installing the 0.4.40 RPM does not work.

**Comment by Bergg** [comment]
sudo dnf update N1-0.4.40.rpm
Last metadata expiration check: 0:03:00 ago on Sun May 22 16:07:09 2016.
Error: nothing provides libgnome-keyring0 needed by nylas-0.4.40-0.1.x86_64
(try to add '--allowerasing' to command line to replace conflicting packages)

**Comment by MikeChav** [comment]
@RayFoss 
I had to downgrade from Nodejs 4 to 0.10, but it installed fine!  I was able to add my account, install extensions, but then, nothing more. There's no interface.
I'm using Fedora 23 (32 bit)

**Comment by mbilker** [comment]
Huh. It shouldn't require a Nodejs downgrade as N1 contains the Nodejs runtime already.

**Comment by tuxtard** [comment]
Still have a dependency problem with 0.4.40 on OpenSUSE:

Problem: nothing provides gir1.2-gnomekeyring-1.0 needed by nylas-0.4.40-0.1.x86_64

**Comment by RayFoss** [comment]
Same issue on latest stable 1.0.1
```
[xxx ~]$ sudo dnf install '/home/amama/Downloads/N.rpm' --allowerasing
Last metadata expiration check: 0:00:27 ago on Mon Jun  6 09:22:09 2016.
Error: nothing provides libgnome-keyring0 needed by nylas-0.4.40-0.1.x86_64
[xxx ~]$
```

**Comment by ozky** [comment]
I can confirm this in Mageia 5.
Right library name is libgnome-keyring0.
nylas-0.4.45-0.1.x86_64 (due to unsatisfied libgnome-keyring)

**Comment by bengotow** [comment]
Hey folks - It sounds like there are multiple issues going on here. The ticket was opened by Fedora / OpenSuse users who reported that the dependency must be renamed from `libgnome-keyring0` to `libgnome-keyring`. We did that and shipped it in 0.4.40. If you're seeing issues with the new release and need `libgnome-keyring0`, you might want to try installing from the `deb` package instead of the RPM. Hope that helps! Please open new issues if the problem persists.

**Comment by Bergg** [comment]
as of nylas-0.4.47-0.1.x86_64 the command dnf update works now for me without the dependency issue.
